source/slang/slang-options.cpp

```diff
@@ -1009,6 +1032,33 @@ struct OptionsParser
                 {
                     requestImpl->getSink()->setFlag(DiagnosticSink::Flag::VerbosePath);
                 }
+                else if (argValue == "-warnings-as-errors")
+                {
+                    CommandLineArg operand;
+                    SLANG_RETURN_ON_FAIL(reader.expectArg(operand));
+
+                    if (operand.value == "all")
+                        requestImpl->getSink()->setFlag(DiagnosticSink::Flag::TreatWarningsAsErrors);
+                    else
+                    {
+                        if (overrideDiagnosticSeverity(operand.value, Severity::Warning, Severity::Error) != SLANG_OK)
```

Comment:
Should really use SLANG_SUCCEEDED/SLANG_FAILED, as opposed to testing for success/failure, because success isn't the same as SLANG_OK (following COM conventions).

Reply:
Done.